pyoozie/oozie_api.py

```diff
@@ -0,0 +1,561 @@
+
+logger = logging.getLogger('pyoozie.OozieAPI')
+logging.getLogger('requests').setLevel(logging.WARNING)
+
+
+class OozieAPI(object):
+
+    JOB_TYPE_STRINGS = {
+        ArtifactType.Coordinator: ('coordinator', 'coordinatorjobs'),
+        ArtifactType.Workflow: ('wf', 'workflows'),
+    }
+
+    JOB_TYPES = {
+        ArtifactType.Coordinator: Coordinator,
+        ArtifactType.CoordinatorAction: CoordinatorAction,
+        ArtifactType.Workflow: Workflow,
+        ArtifactType.WorkflowAction: WorkflowAction,
+    }
+
+    STATUS_TYPES = {
+        ArtifactType.Coordinator: Coordinator.Status,
+        ArtifactType.CoordinatorAction: CoordinatorAction.Status,
+        ArtifactType.Workflow: Workflow.Status,
+        ArtifactType.WorkflowAction: WorkflowAction.Status,
+    }
+
+    class Stats(object):
+
+        def __init__(self):
+            self.reset()
+
+        def reset(self):
+            self._requests = 0
+            self._errors = 0
+            self._bytes_received = 0
+            self._elapsed = 0
+
+        def update(self, response):
+            self._requests += 1
+            if response is not None:
+                if not response:
+                    self._errors += 1
+                self._bytes_received += len(response.text)
+                self._elapsed += response.elapsed.microseconds
+            else:
+                self._errors += 1
+
+        @property
+        def requests(self):
+            return self._requests
+
+        @property
+        def errors(self):
+            return self._errors
+
+        @property
+        def bytes_received(self):
+            return self._bytes_received
+
+        @property
+        def elapsed(self):
+            return self._elapsed
+
+    def __init__(self, url=None, user=None, timeout=None, verbose=True, **_):
+        oozie_url = (url or 'http://localhost').rstrip('/')
+        if not oozie_url.endswith('/oozie'):
+            oozie_url += '/oozie'
+        self._url = oozie_url
+        self._user = user
+        self._timeout = timeout or 30
+        self._verbose = verbose  # Note: change default for verbose!
+        self._stats = OozieAPI.Stats()
+        self._test_connection()
+
+    def _test_connection(self):
+        response = None
+        try:
+            response = requests.get('{}/versions'.format(self._url), timeout=self._timeout)
+            response.raise_for_status()
+            self._stats.update(response)
+        except requests.RequestException as err:
+            self._stats.update(response)
+            if self._verbose and response is not None:
+                logger.error(response.headers)
+            message = "Unable to contact Oozie server at {}".format(self._url)
+            raise OozieException.communication_error(message, err)
+        versions = json.loads(response.content)
+        if 2 not in versions:
+            message = "Oozie server at {} does not support API version 2 (supported: {})".format(self._url, versions)
```

Comment:
As an aside, is there anything that v2 gives us that v1 doesn't? Should we eventually add v2 support?